### PR TITLE
Refactor/volley out retrofit in

### PR DESCRIPTION
**Pull request by mrsasha**
@skraynick chk pls.

### CODE REVIEW

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/TheInternalApplicationBootstrapperShould.java [hunk 1]
```diff
@@ -77,28 +81,30 @@ public void setUp() throws Exception {
     }
 
     @Test
-    public void test_suppression_time() throws Exception {
-        tested.onResolutionsFinished(Arrays.asList(beaconEventSupressionTime));
-        verify(tested, times(1)).presentBeaconEvent(any(BeaconEvent.class));
+    public void beaconEventFilterShouldHaveOneSuppressionTimeEvent() throws Exception {
+        List<BeaconEvent> events = ListUtils.filter(Arrays.asList(beaconEventSupressionTime), tested.beaconEventFilter);
+        Assertions.assertThat(events.size()).isEqualTo(1);
+
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/TheInternalApplicationBootstrapperShould.java]: no space

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/action/ActionFactoryTest.java [hunk 2]
```diff
@@ -31,45 +28,126 @@
...
+            Assertions.assertThat(result).isNotNull();
+            Assertions.assertThat(result).isInstanceOf(InAppAction.class);
+            Assertions.assertThat(((InAppAction) result).getSubject()).isNull();
+            Assertions.assertThat(((InAppAction) result).getBody()).isNull();
+            Assertions.assertThat(((InAppAction) result).getUri().toString()).isEqualTo("http://www.google.com");
+
+        } catch (Exception e) {
+            Assertions.fail(e.getMessage());
+        }
+    }
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/action/ActionFactoryTest.java]: let's take out space.

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/di/TestProvidersModule.java [hunk 3]
```diff
@@ -61,34 +69,49 @@ public TestServiceScheduler provideTestServiceScheduler(Context context, AlarmMa
...
+        return new SettingsManager(transport, sharedPreferences);
+    }
+
+    @Provides
+    @Named("successfulRetrofitApiService")
+    @Singleton
+    public RetrofitApiServiceImpl provideSuccessfulRetrofitApiService(Context context, Gson gson, @Named("testPlatformIdentifier") PlatformIdentifier platformIdentifier) {
+        return new SuccessfulRetrofitApiService(context, gson, platformIdentifier, "http://test.com");
+    }
+
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/di/TestProvidersModule.java]: no space

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/model/BeaconIdTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
+package com.sensorberg.sdk.model;
+
+import org.junit.Test;
+
+import static org.fest.assertions.api.Assertions.assertThat;
+
+
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/model/BeaconIdTest.java]: no space

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/model/BeaconIdTest.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+public class BeaconIdTest {
+
+    BeaconId beacon1_from_byte_array = new BeaconId(new byte[]{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19});
+    BeaconId beacon1_from_byte_array_zero_offset = new BeaconId(new byte[]{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19}, 0);
+    BeaconId beacon1_from_byte_array_with_offset = new BeaconId(new byte[]{0, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19},
+            1);
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/model/BeaconIdTest.java]: one line

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/model/server/TheResolveResponse.java [hunk 6]
```diff
@@ -1,25 +1,72 @@
...
+    @Inject
+    Gson gson;
+
+    private static final JsonObject ANY_IN_APP_JSON = new JsonObject();
+
+    static {
+        try {
+            ANY_IN_APP_JSON.addProperty("url", "sensorberg://");
+        } catch (Exception e) {
+        }
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/model/server/TheResolveResponse.java]: Could be redundant, but if we are going to catch the exception let's have the error message printed...

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/model/sugar/TheSugarScanobjectShould.java [hunk 7]
```diff
@@ -1,7 +1,10 @@
 package com.sensorberg.sdk.model.sugar;
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/model/sugar/TheSugarScanobjectShould.java]: Just noticed the class name is not camal case.

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/action/TestTheInAppAction.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+import android.os.Parcel;
+import android.test.AndroidTestCase;
+
+import java.util.UUID;
+
+
+public class TestTheInAppAction extends AndroidTestCase {
+
+    private static final String MESSAGE = "message";
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/action/TestTheInAppAction.java]: We are using these three static variables Can we not put them in some kind of settings or constant class?

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/model/sugar/TheSugarScanobjectShould.java (hunk 7 above)
Comment (mrsasha) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/model/sugar/TheSugarScanobjectShould.java]: none of the test classes have them. Also, inside tests most of them still have snake case titles. I tried not to touch those, we can do that separately, didn't want to confuse refactor with renaming.

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/internal/http/ApiServiceShould.java [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+        Assertions.assertThat(response1.isSuccessful()).isTrue();
+
+        Call<BaseResolveResponse> call2 = realRetrofitApiService.updateBeaconLayout("");
+        Response<BaseResolveResponse> response2 = call2.execute();
+        Assertions.assertThat(response2.isSuccessful()).isTrue();
+
+        Assertions.assertThat(response2.raw().cacheResponse()).isNotNull();
+        Assertions.assertThat(response2.raw().networkResponse()).isNull();
+    }
+
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/internal/http/ApiServiceShould.java]: no space

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/TheInternalBootstrapperIntegration.java [hunk 10]
```diff
@@ -1,72 +1,81 @@
...
+    private Transport spiedTransportWithMockService;
+
+    private static final JsonObject ANY_IN_APP_JSON = new JsonObject();
 
     static {
         try {
-            ANY_IN_APP_JSON.put("url", "sensorberg://");
-        } catch (JSONException e) {
+            ANY_IN_APP_JSON.addProperty("url", "sensorberg://");
+        } catch (Exception e) {
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/TheInternalBootstrapperIntegration.java]: comment or print exception message.

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/TheInternalBootstrapperIntegration.java [hunk 11]
```diff
@@ -108,132 +117,107 @@
...
-        enqueue(com.sensorberg.sdk.test.R.raw.response_resolve_precaching);
-        Assertions.assertThat(tested.proximityUUIDs).hasSize(0);
+        BaseResolveResponse resolveResponse = gson.fromJson(Utils.getRawResourceAsString(com.sensorberg.sdk.test.R.raw.response_resolve_precaching,
+                InstrumentationRegistry.getContext()), BaseResolveResponse.class);
+        Mockito.when(mockRetrofitApiService.updateBeaconLayout(Mockito.anyString())).thenReturn(Calls.response(resolveResponse));
+
+        Assertions.assertThat(spiedInternalApplicationBootstrapper.proximityUUIDs).hasSize(0);
 
-        tested.updateBeaconLayout();
+        System.out.println("TheInternalBootstrapperIntegration start test_precaching_of_account_proximityUUIDS");
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/TheInternalBootstrapperIntegration.java]: was this only for debugging?

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/action/TestTheInAppAction.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+import android.os.Parcel;
+import android.support.test.runner.AndroidJUnit4;
+
+import java.util.UUID;
+
+@RunWith(AndroidJUnit4.class)
+public class TestTheInAppAction {
+
+    private static final String MESSAGE = "message";
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/action/TestTheInAppAction.java]: We are using these three variables in three places, perhaps we can somehow them somewhere where we can reuse them? (not sure where my original comment went so putting it again!)

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/action/TestTheInAppAction.java (hunk 8 above)
Comment (mrsasha) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/action/TestTheInAppAction.java]: yeah, we should.

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/internal/http/ApiServiceShould.java [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+    @Test
+    public void apiservice_should_have_valid_installationid_in_header() throws Exception {
+        Call<BaseResolveResponse> call = realRetrofitApiService.updateBeaconLayout("");
+        Response<BaseResolveResponse> response = call.execute();
+
+        Assertions.assertThat(response.raw().request().headers()).isNotNull();
+        Assertions.assertThat(response.raw().request().headers().get(Transport.HEADER_INSTALLATION_IDENTIFIER))
+                .isEqualTo(realPlatformIdentifier.getDeviceInstallationIdentifier());
+
```
Comment (skraynick) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/internal/http/ApiServiceShould.java]: no space

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/TheInternalBootstrapperIntegration.java (hunk 11 above)
Comment (mrsasha) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/TheInternalBootstrapperIntegration.java]: yeah, gonna remove it. good catch.

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/model/server/TheResolveResponse.java (hunk 6 above)
Comment (mrsasha) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/model/server/TheResolveResponse.java]: ok

File: android-sdk/src/androidTest/java/com/sensorberg/sdk/action/TestTheInAppAction.java (hunk 12 above)
Comment (mrsasha) [inline android-sdk/src/androidTest/java/com/sensorberg/sdk/action/TestTheInAppAction.java]: I can still see the comment, it's just hidden. I'm not pushing now so it doesn't hide the other comments, gonna wait for you to finish the PR.

File: android-sdk/src/main/java/com/sensorberg/sdk/internal/transport/RetrofitApiTransport.java [hunk 14]
```diff
@@ -0,0 +1,221 @@
...
+
+    @Override
+    public void setLoggingEnabled(boolean enabled) {
+        getApiService().setLoggingEnabled(enabled);
+    }
+
+    public <T> void enqueueWithRetry(Call<T> call, final Callback<T> callback) {
+        call.enqueue(new CallbackWithRetry<>(callback));
+    }
+
```
Comment (skraynick) [inline android-sdk/src/main/java/com/sensorberg/sdk/internal/transport/RetrofitApiTransport.java]: no space

File: android-sdk/src/main/java/com/sensorberg/sdk/internal/transport/interfaces/RetrofitApiService.java [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+
+    @GET
+    Call<ResolveResponse> getBeacon(@Url String beaconURLString, @Header("X-pid") String beaconId, @Header("X-qos") String networkInfo);
+
+    @POST
+    Call<ResolveResponse> publishHistory(@Url String beaconLayoutUrl, @Body HistoryBody body);
+
+    @GET
+    Call<SettingsResponse> getSettings(@Url String settingsUrl);
+
```
Comment (skraynick) [inline android-sdk/src/main/java/com/sensorberg/sdk/internal/transport/interfaces/RetrofitApiService.java]: no space

File: android-sdk/src/main/java/com/sensorberg/sdk/internal/transport/model/SettingsResponse.java [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+    @Getter
+    private com.sensorberg.sdk.settings.Settings settings;
+
+    public SettingsResponse() {}
+
+    public SettingsResponse(int rev, com.sensorberg.sdk.settings.Settings set) {
+        revision = rev;
+        settings = set;
+    }
+
```
Comment (skraynick) [inline android-sdk/src/main/java/com/sensorberg/sdk/internal/transport/model/SettingsResponse.java]: no space

**Comment by skraynick** [comment]
@mrsasha done my review. The branch apparently has conflicts...

**Comment by mrsasha** [comment]
ok, cool, now I'm gonna rebase/merge, gonna see what works better.

**Comment by mrsasha** [comment]
@skraynick  Ok, fixed everything, tests pass, travis builds. Whenever you're ready. :)

**Pull request merged**